compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -215,9 +260,65 @@ bool TaggedShapeAnalyzer::verify_tag() const
   return true;
 }
 
+/**
+ * @brief Initialize the class members and check under conditions
+ *
+ *  Condtiions that have to be met for analyzer
+ *    c1: input rank >= output rank
+ *    c2: The 'perm' of tranpose should be a CircleConst* type
+ *    c3: The shapes of the given nodes should be all known
+ *
+ * @return True, if all conditions are satisfied and class members are initialized successfully
+ *         False, otherwise
+ */
+template <loco::DataType DType>
+bool TaggedShapeAnalyzer::init(const luci::CircleTranspose *front_transpose,
+                               const luci::CircleReshape *mid_reshape,
+                               const luci::CircleTranspose *back_transpose)
+{
+  _in = dynamic_cast<luci::CircleNode *>(front_transpose->a());
+  _front_transpose = front_transpose;
+  _mid_reshape = mid_reshape;
+  _back_transpose = back_transpose;
+
+  // check c1
+  RET_FALSE_UNLESS(_in->rank() >= _back_transpose->rank());
+
+  const auto front_perm = dynamic_cast<luci::CircleConst *>(_front_transpose->perm());
+  const auto back_perm = dynamic_cast<luci::CircleConst *>(_back_transpose->perm());
+
+  // check c2
+  RET_FALSE_UNLESS(front_perm != nullptr);
+  RET_FALSE_UNLESS(front_perm->dtype() == DType);
+  RET_FALSE_UNLESS(back_perm != nullptr);
+  RET_FALSE_UNLESS(back_perm->dtype() == DType);
+
+  _in_shape = extract_shape(_in);
+  _front_perm = extract_const<DType>(front_perm);
+  _mid_shape = extract_shape(_mid_reshape);
+  _back_perm = extract_const<DType>(back_perm);
+
+  auto all_known = [](const std::vector<int32_t> &v) -> bool {
+    for (auto i : v)
+      if (i <= 0)
+        return false;
+    return true;
+  };
+
+  // check c3
+  RET_FALSE_UNLESS(all_known(_in_shape));
+  RET_FALSE_UNLESS(all_known(extract_shape(_front_transpose)));
+  RET_FALSE_UNLESS(all_known(_mid_shape));
+  RET_FALSE_UNLESS(all_known(extract_shape(_back_transpose)));
+
+  return true;
+}
+
 /**
  * @brief check 'Transpose-Reshape-Transpose' can be replaced by one 'Reshape'.
  *
+ * @warning '@init' have to be called first
```

Comment:
Could you add `assert`s for this condition?

It would be enough to check private variables `_in`, `_front_XX`, ... are not nullptr.